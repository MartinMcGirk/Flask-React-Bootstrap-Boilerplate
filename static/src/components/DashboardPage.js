import React, { Component } from 'react'
import { connect } from 'react-redux';
import { startLoadUsers } from '../actions/users';
import { CandidateDashboard } from './candidate/CandidateDashboard';
import { RecruiterDashboard } from './recruiter/RecruiterDashboard';

class DashboardPageComponent extends Component {
    componentDidMount() {
        this.props.startLoadUsers(this.props.token)
    }

    getRoleSpecificDashboard = (role) => {
        if (role === 'candidate') {
            return <CandidateDashboard></CandidateDashboard>
        } else if (role === 'recruiter') {
            return <RecruiterDashboard></RecruiterDashboard>
        }
        return <div>Other dashboard</div>
    }

    render() {
        return (
            <div>
                { this.getRoleSpecificDashboard(this.props.user.role) }
            </div>
        )
    }
}

const mapStateToProps = (state) => ({
    token: state.auth.token,
    user: state.auth.user,
});

const mapDispatchToProps = (dispatch) => ({
    startLoadUsers: (token) => dispatch(startLoadUsers(token))
});

const DashboardPage = connect(mapStateToProps, mapDispatchToProps)(DashboardPageComponent);

export { DashboardPage }