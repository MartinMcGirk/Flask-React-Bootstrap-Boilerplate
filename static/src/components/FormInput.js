import React from 'react'

const FormInput = ({
    fieldId,
    label,
    placeholder,
    value,
    onFieldChange,
    smallText,
    inputType='text',
    classExtras = ''
}) => (
    <div className={`form-group ${classExtras}`}>
        <label htmlFor={`${fieldId}Field`}>{label}</label>
        <input
            type={inputType}
            className="form-control"
            id={`${fieldId}Field`}
            placeholder={placeholder}
            value={value}
            onChange={onFieldChange}
        />
        {smallText && <small id="emailHelp" className="form-text text-muted">{smallText}</small>}
    </div>
);

export { FormInput }