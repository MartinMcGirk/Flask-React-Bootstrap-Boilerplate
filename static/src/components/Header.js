import React from 'react'
import { NavLink } from 'react-router-dom';
import { startLogout } from '../actions/auth';
import { connect } from 'react-redux';
import { getRoleFromToken } from '../utils/misc';

const Header = ({ startLogout, role }) => (
    <nav className="navbar navbar-expand-lg navbar-dark bg-primary">
        <NavLink className="navbar-brand mb-0 h1" to="/dashboard">
            Boilerplate
        </NavLink>
        <button
            className="navbar-toggler"
            type="button"
            data-toggle="collapse"
            data-target="#navbarSupportedContent"
            aria-controls="navbarSupportedContent"
            aria-expanded="false"
            aria-label="Toggle navigation"
        >
            <span className="navbar-toggler-icon"/>
        </button>
        <div className="collapse navbar-collapse" id="navbarSupportedContent">
            <ul className="navbar-nav mr-auto">
                <li className="nav-item">
                    <NavLink className="nav-link" to="/dashboard" exact={true}>
                        Dashboard
                    </NavLink>
                </li>
                {
                    role === 'recruiter' ? (
                        <li className="nav-item d-md-none">
                            <NavLink className="nav-link" to="/profile">
                                Profile
                            </NavLink>
                        </li>
                    ) : (
                        <li className="nav-item">
                            <NavLink className="nav-link" to="/profile">
                                Profile
                            </NavLink>
                        </li>
                    )
                }
                {
                    role === 'recruiter' && (
                        <li className="nav-item d-md-none">
                            <NavLink className="nav-link" to="/company">
                                Company
                            </NavLink>
                        </li>
                    )
                }
                {
                    role === 'recruiter' && (
                        <li className="nav-item d-md-none">
                            <NavLink className="nav-link" to="/vacancies">
                                Vacancies
                            </NavLink>
                        </li>
                    )
                }
                {
                    role === 'recruiter' && (
                        <li className="nav-item d-md-none">
                            <NavLink className="nav-link" to="/candidates">
                                Candidates
                            </NavLink>
                        </li>
                    )
                }
            </ul>
            <ul className="navbar-nav">
                <li className="nav-item">
                    <button className="btn btn-outline-light" onClick={startLogout}>Logout</button>
                </li>
            </ul>
        </div>
    </nav>
);

const mapDispatchToProps = (dispatch) => ({
    startLogout: () => dispatch(startLogout())
});

const mapStateToProps = (state) => ({
    role: getRoleFromToken(state.auth.token)
});

const ConnectedHeader = connect(mapStateToProps, mapDispatchToProps)(Header);

export { Header, ConnectedHeader }