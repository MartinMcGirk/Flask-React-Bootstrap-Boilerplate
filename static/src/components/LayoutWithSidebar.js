import React from 'react'
import { Sidebar } from './Sidebar';

const LayoutWithSidebar = (props) => {
    const SidebarComponent = props.Sidebar || Sidebar;
    return (
        <div className="container-fluid">
            <div className="row">
                <div className="col-md-2 d-none d-md-block">
                    <SidebarComponent>

                    </SidebarComponent>
                </div>
                <div className="col-sm-12 col-md-10">
                    {props.children}
                </div>
            </div>
        </div>
    );
};



export { LayoutWithSidebar }