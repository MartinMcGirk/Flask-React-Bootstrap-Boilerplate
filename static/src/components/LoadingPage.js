import React from 'react'

const LoadingPage = () => (
    <div className="loader">
        <img className="loader__image" src="/images/loader.gif" alt="Loading gif"/>
    </div>
);

export { LoadingPage }