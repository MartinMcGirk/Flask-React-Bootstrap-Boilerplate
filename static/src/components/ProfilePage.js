import React, { Component } from 'react'
import { connect } from 'react-redux';
import { CandidateProfile } from './candidate/CandidateProfile';

class ProfilePageComponent extends Component {
    getRoleSpecificDashboard = (role) => {
        if (role === 'candidate') {
            return <CandidateProfile></CandidateProfile>
        } else if (role === 'recruiter') {
            return <div></div>
        }
        return <div>Other Profile</div>
    };

    render() {
        return (
            <div>
                { this.getRoleSpecificDashboard(this.props.user.role) }
            </div>
        )
    }
}

const mapStateToProps = (state) => ({
    user: state.auth.user,
});

const ProfilePage = connect(mapStateToProps)(ProfilePageComponent);

export { ProfilePage }