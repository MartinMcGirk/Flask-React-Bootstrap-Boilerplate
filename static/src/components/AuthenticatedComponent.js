import React from 'react';
import { connect } from 'react-redux';
import { history } from '../routers/AppRouter';
import { checkTokenIsValid } from '../comms/backend';
import { loginFailure, loginSuccess } from '../actions/auth';
import { parseJSON } from '../utils/misc';
import { ConnectedHeader } from './Header';

function mapStateToProps(state) {
    return {
        token: state.auth.token,
        isAuthenticated: state.auth.isAuthenticated,
    };
}

const mapDispatchToProps = (dispatch) => ({
    loginSuccess: (token) => dispatch(loginSuccess(token)),
    loginFailure: () => dispatch(loginFailure())
});

export function requireAuthentication(Component) {
    class AuthenticatedComponent extends React.Component {
        componentWillMount() {
            this.setState(() => ({
                loaded_if_needed: false,
            }));
            this.checkAuth();
        }

        componentWillReceiveProps(nextProps) {
            this.checkAuth(nextProps);
        }

        checkAuth(props = this.props) {
            if (!props.isAuthenticated) {
                const token = localStorage.getItem('token');
                if (!token) {
                    history.push('/login');
                } else {
                    checkTokenIsValid(token)
                        .then(parseJSON)
                        .then(res => {
                            if (res.token_is_valid) {
                                this.props.loginSuccess(token);
                                this.setState(() => ({
                                    loaded_if_needed: true,
                                }));

                            } else {
                                history.push('/login');
                            }
                        });

                }
            } else {
                this.setState(() => ({
                    loaded_if_needed: true
                }));
            }
        }

        render() {
            return (
                <div>
                    {this.props.isAuthenticated && this.state.loaded_if_needed
                        ? (
                            <div>
                                <ConnectedHeader />
                                <Component {...this.props} />
                            </div>
                        )
                        : null
                    }
                </div>
            );

        }
    }

    return connect(mapStateToProps, mapDispatchToProps)(AuthenticatedComponent);
}