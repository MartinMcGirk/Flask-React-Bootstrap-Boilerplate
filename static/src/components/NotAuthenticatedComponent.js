import React from 'react';
import { connect } from 'react-redux';
import { checkTokenIsValid } from '../comms/backend';
import { parseJSON } from '../utils/misc';
import { loginFailure, loginSuccess } from '../actions/auth';
import { history } from '../routers/AppRouter';
import { ConnectedHeader } from './Header';
import { PublicHeader } from './static/PublicHeader';

function mapStateToProps(state) {
    return {
        token: state.auth.token,
        isAuthenticated: state.auth.isAuthenticated,
    };
}

const mapDispatchToProps = (dispatch) => ({
    loginSuccess: (token) => dispatch(loginSuccess(token)),
    loginFailure: () => dispatch(loginFailure())
});

export function requireNoAuthentication(Component) {

    class NotAuthenticatedComponent extends React.Component {

        constructor(props) {
            super(props);
            this.state = {
                loaded: false,
            };
        }

        componentWillMount() {
            this.checkAuth();
        }

        componentWillReceiveProps(nextProps) {
            this.checkAuth(nextProps);
        }

        checkAuth(props = this.props) {
            if (props.isAuthenticated) {
                history.push('/dashboard');

            } else {
                const token = localStorage.getItem('token');
                if (token) {
                    checkTokenIsValid(token)
                        .then(parseJSON)
                        .then(res => {
                            if (res.token_is_valid) {
                                this.props.loginSuccess(token);
                                history.push('/dashboard');

                            } else {
                                this.setState({
                                    loaded: true,
                                });
                            }
                        });
                } else {
                    this.setState({
                        loaded: true,
                    });
                }
            }
        }

        render() {
            return (
                <div>
                    {!this.props.isAuthenticated && this.state.loaded
                        ? (
                            <div>
                                <PublicHeader />
                                <Component {...this.props} />
                            </div>
                        )
                        : null
                    }
                </div>
            );

        }
    }

    return connect(mapStateToProps, mapDispatchToProps)(NotAuthenticatedComponent);

}