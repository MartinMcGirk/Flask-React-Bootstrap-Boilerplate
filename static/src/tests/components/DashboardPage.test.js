import React from 'react';
import { shallow } from 'enzyme';
import { DashboardPage } from '../../components/DashboardPage';

test('should correctly render the expense dashboard page', () => {
    const wrapper = shallow(<DashboardPage/>);
    expect(wrapper).toMatchSnapshot();
});