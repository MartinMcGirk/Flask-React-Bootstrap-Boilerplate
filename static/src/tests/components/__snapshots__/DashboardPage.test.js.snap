// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render the expense dashboard page 1`] = `
<div>
  Dashboard page content
</div>
`;
